src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -138,5 +141,52 @@ def calculate_prices(key, functor):
                 self._update_tree_summary(child)
 
     def _update_bills(self, area, past_market_types):
-        result = energy_bills(area, past_market_types)
-        return OrderedDict(sorted(result.items()))
+        result, external_trades = energy_bills(area, past_market_types, {})
+        return OrderedDict(sorted(result.items())), external_trades
+
+    def _calculate_redis_bills(self, area, energy_bills, external_trades):
+        from copy import deepcopy
+        flattened = self._flatten_energy_bills(deepcopy(energy_bills), {})
+        return self._accumulate_by_children(area, flattened, {}, external_trades)
+
+    def _flatten_energy_bills(self, energy_bills, flat_results):
+        for k, v in energy_bills.items():
+            if "children" in v:
+                self._flatten_energy_bills(v["children"], flat_results)
+            flat_results[k] = v
+            flat_results[k].pop("children", None)
+        return flat_results
+
+    def _accumulate_by_children(self, area, flattened, results, external_trades):
+        if not area.children:
+            # This is a device
+            results[area.uuid] = flattened[area.name]
+        else:
+            results[area.uuid] = [
+                {c.name: flattened[c.name]}
+                for c in area.children
+            ]
+            results.update(**self._generate_external_and_total_bills(area, results, flattened))
+
+            for c in area.children:
+                results.update(
+                    **self._accumulate_by_children(c, flattened, results, external_trades)
+                )
+        return results
+
+    def _generate_external_and_total_bills(self, area, results, flattened):
+
+        all_child_results = [v for i in results[area.uuid] for _, v in i.items()]
+        results[area.uuid].append({"Accumulated Trades": {
+            'bought': sum(v['bought'] for v in all_child_results),
+            'sold': sum(v['sold'] for v in all_child_results),
+            'spent': sum(v['spent'] for v in all_child_results),
+            'earned': sum(v['earned'] for v in all_child_results),
+            'total_energy': sum(v['total_energy'] for v in all_child_results),
+            'total_cost': sum(v['total_cost'] for v in all_child_results),
```

Comment:
Not that i find it very important, but i dont see a member "type" here. Is this on purpose?

Reply:
Forget it, for the accumulated_bills for the area a type is not needed, right?

Reply:
Just to give some feedback on that, the type member is actually only needed for the former frontend, in order to know when to render a house, and when to render a cell tower image. Now that this is no longer the case, I think it is better to get rid of it globally. If you agree I will completely remove it.

Reply:
Yes, please remove. I can not think of a usage later on. Thanks

Reply:
Thanks, already deleted this.